src/h5web/vis-packs/nexus/models.ts

```diff
@@ -26,17 +26,19 @@ export enum NxInterpretation {
   Image = 'image',
 }
 
-export interface NxData {
-  signalDataset: Dataset<ArrayShape, NumericType | ComplexType>;
+export interface NxData<
+  T extends NumericType | ComplexType = NumericType | ComplexType
+> {
+  signalDataset: Dataset<ArrayShape, T>;
   errorsDataset?: NumArrayDataset;
```

Comment:
In the end, I kept `NumArrayDataset` for `errorsDatasets` and `auxDatasets`. I had issues with the conditional type :confused: